snapm/fsdiff/cache.py

```
@@ -0,0 +1,279 @@
+        cache_path = os.path.join(_DIFF_CACHE_DIR, file_name)
+        if not valid:
+            _log_info("Pruning expired cache file %s", file_name)
+            try:
+                os.unlink(cache_path)
+            except OSError as err:
+                _log_error("Error unlinking stale cache file %s: %s", file_name, err)
+            continue
+
+        if str(uuid_a) == load_uuid_a and str(uuid_b) == load_uuid_b:
+            with lzma.open(os.path.join(_DIFF_CACHE_DIR, file_name), "rb") as fp:
+                candidate = pickle.load(fp)
```

Ruff (0.14.8)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)